Source/SwiftLintFramework/Rules/Idiomatic/FatalErrorMessageRule.swift

```diff
@@ -36,32 +36,38 @@ public struct FatalErrorMessageRule: ASTRule, ConfigurationProviderRule, OptInRu
         ]
     )
 
-    public func validate(file: SwiftLintFile, kind: SwiftExpressionKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        guard kind == .call,
-            let offset = dictionary.offset,
-            dictionary.name == "fatalError",
-            hasEmptyBody(dictionary: dictionary, file: file) else {
-                return []
-        }
-
-        return [
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, byteOffset: offset))
-        ]
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor()
     }
+}
+
+private extension FatalErrorMessageRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
 
-    private func hasEmptyBody(dictionary: SourceKittenDictionary, file: SwiftLintFile) -> Bool {
-        guard let bodyRange = dictionary.bodyByteRange else {
-            return false
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            guard let expression = node.calledExpression.as(IdentifierExprSyntax.self),
+                expression.identifier.text == "fatalError",
+                node.argumentList.isEmptyOrEmptyString else {
+                return
+            }
+
+            violationPositions.append(node.positionAfterSkippingLeadingTrivia)
         }
+    }
+}
 
-        if bodyRange.length == 0 {
+private extension TupleExprElementListSyntax {
```

Comment:
at what point should we start extracting these extensions? this was copied from `XCTFailMessageRule`